core/frontend/src/quantity-formatting/BaseUnitFormattingSettingsProvider.ts

```diff
@@ -0,0 +1,149 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module Utils
+ */
+
+import { FormatProps } from "@bentley/imodeljs-quantity";
+import { IModelApp } from "../IModelApp";
+import { IModelConnection } from "../IModelConnection";
+import { SelectedViewportChangedArgs } from "../ViewManager";
+import {
+  FormattingUnitSystemChangedArgs, OverrideFormatEntry, QuantityFormatOverridesChangedArgs, QuantityFormatter,
+  QuantityTypeKey, UnitFormattingSettingsProvider, UnitSystemKey,
+} from "./QuantityFormatter";
+
+/** This abstract class can monitor active imodel changes and update the QuantityFormatter overrides and the active unit system based on

```

Comment:
Found the wording of the docs a bit hard to follow. What do you think of something like this?

```md
This abstract class reacts to changes in the currently active iModel and updates the [[QuantityFormatter]] overrides and active unit system based on stored preferences.

In addition, monitors the [[QuantityFormatter]] for changes to format overrides and the active unit system and stores these changes.
```

Also, do we qualify what an "active iModel" is anywhere else? If not, maybe add a note about what it means in this context.

Reply:
Updated.